ci-operator/step-registry/ipi/conf/gcp/windows-machineset/ipi-conf-gcp-windows-machineset-commands.sh

```diff
@@ -10,7 +10,7 @@ oc wait deployment windows-machine-config-operator -n openshift-windows-machine-
 # Ensure userDataSecret exist, fail otherwise. The userDataSecret is required and contains specific information to
 # customize the machine at first boot. For instance, the authorized public key for the SSH server to accept
 # incoming connections, firewall rules, etc.
-oc -n openshift-machine-api get secret "${WINDOWS_USER_DATA_SECRET}"
+until oc -n openshift-machine-api get secret "${WINDOWS_USER_DATA_SECRET}" 2> /dev/null; do echo -n "." && sleep 5; done
```

Comment:
Consider increasing the sleep time. For what it's worth, WMCO waits 15 seconds before retrying API calls.

Reply:
Also, if you want to fail fast here (rather than waiting on the entire job's timeout), consider using the [timeout command](http://manpages.courier-mta.org/htmlman1/timeout.1.html). example:
```
timeout 3m bash -c 'until oc -n openshift-machine-api get secret "${WINDOWS_USER_DATA_SECRET}" 2> /dev/null; do echo -n "." && sleep 15; done'
```